### PR TITLE
fix(WebWorkers): Run XHR requests on the UI

### PR DESCRIPTION
**Pull request by jteplitz**
Fixes issues in dart where dart:html is not available in isolates and
allows for better profiling of XHR requests

### CODE REVIEW

File: modules/angular2/src/web-workers/ui/impl.ts [hunk 1]
```diff
@@ -207,6 +208,17 @@ export class WebWorkerMain {
     }
   }
 
+  private _handleXhrMessage(data: ReceivedMessage) {
+    var args = data.args;
+    switch (data.method) {
+      case "get":
```
Comment (vicb) [inline modules/angular2/src/web-workers/ui/impl.ts]: handle other methods (by throwing ?) ?

File: modules/angular2/test/web-workers/worker/xhr_impl_spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+         });
+       }));
+  });
+}
+
+@proxy
+@IMPLEMENTS(MessageBroker)
+class SpyMessageBroker extends SpyObject {
+  constructor() { super(MessageBroker) }
+  noSuchMethod(m) { return super.noSuchMethod(m) }
```
Comment (vicb) [inline modules/angular2/test/web-workers/worker/xhr_impl_spec.ts]: Add `;` in the two methods above ?
Is there a new "lint" task to check this ? (I remember having seen something while browsing the Travis log last month)

File: modules/angular2/test/web-workers/worker/xhr_impl_spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import {MessageBroker, UiArguments} from 'angular2/src/web-workers/worker/broker';
+import {WebWorkerXHRImpl} from "angular2/src/web-workers/worker/xhr_impl";
+import {PromiseWrapper} from "angular2/src/facade/async";
+
+export function main() {
+  describe("WebWorkerXHRImpl", () => {
+    it("should pass requests through the broker and return the response",
+       inject([AsyncTestCompleter], (async) => {
+         var URL = "http://www.example.com/test";
+         var RESPONSE = "Example response text";
```
Comment (vicb) [inline modules/angular2/test/web-workers/worker/xhr_impl_spec.ts]: could those be `const` ?

File: modules/angular2/src/web-workers/worker/xhr_impl.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import {Injectable} from 'angular2/di';
+import {Promise} from 'angular2/src/facade/async';
+import {XHR} from 'angular2/src/render/xhr';
+import {FnArg, UiArguments, MessageBroker} from 'angular2/src/web-workers/worker/broker';
+
```
Comment (yjbanov) [inline modules/angular2/src/web-workers/worker/xhr_impl.ts]: Let's explain what this implementation does.

File: modules/angular2/src/web-workers/ui/impl.ts [hunk 5]
```diff
@@ -217,6 +229,8 @@ export class WebWorkerMain {
         return this._handleCompilerMessage(data);
       case "renderer":
         return this._handleRendererMessage(data);
+      case "xhr":
+        return this._handleXhrMessage(data);
```
Comment (yjbanov) [inline modules/angular2/src/web-workers/ui/impl.ts]: Let's leave a TODO to break these up into message bus channels (users should be able to add their own channels) and log an issue to add channels, if we haven't already. Here's an example channel API: https://github.com/yjbanov/barista/blob/master/lib/messaging/isolate.dart#L62

File: modules/angular2/test/web-workers/worker/xhr_impl_spec.ts (hunk 2 above)
Comment (yjbanov) [inline modules/angular2/test/web-workers/worker/xhr_impl_spec.ts]: Someone give @vicb a real task so he doesn't have to browser our Travis logs ;)

File: modules/angular2/src/web-workers/ui/impl.ts (hunk 5 above)
Comment (jteplitz602) [inline modules/angular2/src/web-workers/ui/impl.ts]: Good idea. Made an issue here (https://github.com/angular/angular/issues/3661).

File: modules/angular2/src/web-workers/ui/impl.ts (hunk 1 above)
Comment (jteplitz602) [inline modules/angular2/src/web-workers/ui/impl.ts]: Good call. Currently render only supports get, so I've added a default case that throws.

**Pull request closed without merging**